cinder/tests/unit/scheduler/test_sap_fcd_filter.py

```diff
@@ -0,0 +1,132 @@
+                                      {'capabilities': {},
+                                       'vendor_name': VMWARE_VENDOR,
+                                       'storage_protocol': 'vstorageobject'})
+        self.assertTrue(self.filt_cls._is_vmware_fcd(host))
+
+    def test_is_not_vmware_fcd(self):
+        host = fakes.FakeBackendState('host1@backend1#pool1',
+                                      {'capabilities': {},
+                                       'vendor_name': 'Not VMware',
+                                       'storage_protocol': 'vmdk'})
+        self.assertFalse(self.filt_cls._is_vmware_fcd(host))
+
+    def test_passes_not_migrating(self):
+        host = fakes.FakeBackendState('host1@backend1#pool1',
+                                      {'capabilities': {},
+                                       'vendor_name': VMWARE_VENDOR,
+                                       'storage_protocol': 'vstorageobject'})
+        self.assertTrue(self.filt_cls.backend_passes(host, self.props))
+
+    def test_passes_migrating_to_same_backend_no_pool(self):
+        """This should pass the pool is the same as the original pool"""
+        host = fakes.FakeBackendState('host1@backend1#pool1',
+                                      {'capabilities': {},
+                                       'vendor_name': VMWARE_VENDOR,
+                                       'storage_protocol': 'vstorageobject'})
+        props = self.props.copy()
+        props['request_spec']['destination_host'] = 'host1@backend1'
+        props['request_spec']['operation'] = 'migrate_volume'
+        self.assertTrue(self.filt_cls.backend_passes(host, props))
+
+    def test_migrating_to_same_backend_different_pool_no_pool(self):
+        """This should pass because the destination host provides a pool."""
+        host = fakes.FakeBackendState('host1@backend1#pool2',
+                                      {'capabilities': {},
+                                       'vendor_name': VMWARE_VENDOR,
+                                       'storage_protocol': 'vstorageobject'})
+        props = self.props.copy()
+        props['request_spec']['destination_host'] = 'host1@backend1'
+        props['request_spec']['operation'] = 'migrate_volume'
+        self.assertTrue(self.filt_cls.backend_passes(host, props))
+
+    def test_passes_migrating_to_different_backend_no_pool(self):
+        """This should pass the pool is the same as the original pool."""
+        host = fakes.FakeBackendState('host1@backend2#pool1',
+                                      {'capabilities': {},
+                                       'vendor_name': VMWARE_VENDOR,
+                                       'storage_protocol': 'vstorageobject'})
+        props = self.props.copy()
+        props['request_spec']['destination_host'] = 'host1@backend2'
+        props['request_spec']['operation'] = 'migrate_volume'
+        self.assertTrue(self.filt_cls.backend_passes(host, props))
```

Comment:
Aren't these the same tests? I think the second test needs a different host? Not sure given the comment, though.

Reply:
yes